libevmasm/Assembly.cpp

```diff
@@ -756,3 +759,31 @@ LinkerObject const& Assembly::assemble() const
 	}
 	return ret;
 }
+
+vector<size_t> Assembly::decodeSubIds(u256 _data)
+{
+	size_t length = static_cast<size_t>((_data >> 248) & 0xFF);
+	if (length == 0)
+		length = 1;
+	std::vector<size_t> subIds;
+	u256 const mask = 0xFF;
+	while (length--)
+	{
+		auto currentSubId = static_cast<size_t>(_data & mask);
+		subIds.emplace_back(currentSubId);
+		_data >>= 8;
+	}
+	return subIds;
+}
+
+u256 Assembly::encodeSubIds(std::vector<size_t> const& _subIds)
+{
+	u256 encodedSubId = u256(_subIds.size()) << 248;
```

Comment:
This is inconsistent with `decodeSubIds()`, which expects the length byte to be counted too.

Also, I think that `encodedSubIds` would be a better name since the variable does not hold just a single ID.

```suggestion
	u256 encodedSubIds = u256(_subIds.size() + 1) << 248;
```